diffkemp/output.py

```diff
@@ -0,0 +1,141 @@
+from diffkemp.semdiff.caching import ComparisonGraph
+from diffkemp.utils import get_end_line, EndLineNotFound
+from diffkemp.semdiff.result import Result
+import os
+import yaml
+
+
+class YamlOutput:
+    """Yaml representation of compare result."""
+
+    def __init__(self, snapshot_dir_old, snapshot_dir_new, result):
+        self.old_dir = os.path.join(
+            os.path.abspath(snapshot_dir_old), "")
```

Comment:
Isn't `snapshot_dir_old` already an abspath? I think that `YamlOutput` shouldn't do the path transformation, it should be the caller's responsibility.

Reply:
Thanks. The received `snapshot_dir_old` was a relative path (argument of the program). I moved the path transformation to the caller function (`compare`).

I also moved path transformation in the `Callstack` class in methods `as_str_with_rel_paths` and `to_output_yaml_with_rel_path`. In the case of  `as_str_with_rel_paths` I moved it to the caller function (`print_syntax_diff`), in the case of `to_output_yaml_with_rel_path` I removed it completely because the received path was already absolute.